File: techstars-api/spec/factories/companies.rb
FactoryBot.define do
  factory :company do
    name { Faker::Company.name }
    city { Faker::Address.city }
    state { Faker::Address.state_abbr }
    description { Faker::Lorem.paragraphs }
    date_founded { Faker::Date.backward(365) }
  end
end

File: back-end/schemas/companies.js
const Joi = require('joi');

const createCompanySchema = Joi.object({
    name        : Joi.string().trim().required().label('Company Name'),
    description : Joi.string().trim().required().label('Company Description'),
    city        : Joi.string().trim().required().label('Company City'),
    state       : Joi.string().trim().required().label('Company State'),
    date_founded: Joi.date().optional().allow(null).allow("").default(null).label('Company Date Founded'),
});

module.exports = { 
    createCompanySchema,
}